### PR TITLE
doc: clearly explain commit message requirements

### PR DESCRIPTION
**Pull request by romani**
continuation of #5576

### CODE REVIEW

**Comment by romani** [comment]
@rnveach , please review.

File: src/xdocs/contributing.xml [hunk 1]
```diff
@@ -127,26 +127,27 @@
...
 
       <p>
-        Please use <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
-        feature of <a href="https://github.com/">Github</a>.<br/>
-        Please provide wide description of update with detailed explanation of problem
-        and how you propose to resolve it.
-        <b>ATTENTION:</b> Please recheck that in your Pull Request there is ONLY your changes and
-        all of them are rebased from most recent HEAD of our master branch.
+        Please create a <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
+        for your contribution. In the pull request description add explanation of the implementation
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: `pull request` => `Pull Request`
add comma after `description`.
`add explanation` => `add any explanations`

File: src/xdocs/contributing.xml [hunk 2]
```diff
@@ -127,26 +127,27 @@
...
       <p>
-        <b>ATTENTION:</b> Please provide single-line meaningful message for commit with reference
-        to github issue or Pull Request for more details.<br/>
-        Ideal format for message is "Issue #Number: Brief single-line message ..... .". Where
-        NUMBER is Issue or Pull Request number at github.
-        <a href="https://github.com/checkstyle/checkstyle/commit/9303fd9d971eb55cdfd62686ba2f5698edb2c83e">Good example</a>
+        <b>ATTENTION:</b>The commit message must adhere to a certain format.
+         It should be "Issue #Number: Brief single-line message", where NUMBER is the issue number at Github
+         (see <a href="https://github.com/checkstyle/checkstyle/commit/9303fd9d971eb55cdfd62686ba2f5698edb2c83e">an example</a>).
+         Small changes of configuration files, documentation fixes etc. can be contributed by starting
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: add comma before `etc.`

File: src/xdocs/contributing.xml [hunk 3]
```diff
@@ -127,26 +127,27 @@
...
+         the commit message with one of "minor:", "config:", "infra:", "doc:" or "spelling:",
+         followed by a blank and a brief single-line message.
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: `blank` => `space`?

File: src/xdocs/contributing.xml [hunk 4]
```diff
@@ -127,26 +127,27 @@
...
       </p>
 
       <p>
-        After you do Pull Request please recheck that where are no problems from our build systems:
-        <a href="https://travis-ci.org/checkstyle/checkstyle">Travis</a> and others ... .
-        You will see that your Pull Request on Github is marked by that jobs automatically in few minutes.
+        After submitting a pull request, it will be automatically checked by
+        <a href="https://travis-ci.org/checkstyle/checkstyle">Travis</a> and some other continuous integration tools.
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: remove `some`.

File: src/xdocs/contributing.xml [hunk 5]
```diff
@@ -127,26 +127,27 @@
...
+        Therefore please recheck after some minutes that the system didn't find any issues.
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: add comma after `Therefore`.

after `issues` on the end, I think we should add `with your changes`.

File: src/xdocs/contributing.xml [hunk 6]
```diff
@@ -127,26 +127,27 @@
...
+        If there are issues, please fix them and provide an updated version of the same pull request.
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: > please fix them

This should be expanded on by saying to amend the previous commit. Lots of people give us fixes just by adding a second commit.

File: src/xdocs/contributing.xml [hunk 7]
```diff
@@ -127,26 +127,27 @@
...
       <p>
-        Please use <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
-        feature of <a href="https://github.com/">Github</a>.<br/>
-        Please provide wide description of update with detailed explanation of problem
-        and how you propose to resolve it.
-        <b>ATTENTION:</b> Please recheck that in your Pull Request there is ONLY your changes and
-        all of them are rebased from most recent HEAD of our master branch.
+        Please create a <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
+        for your contribution. In the pull request description add explanation of the implementation
+        nuances. Please read Pull Request template for more requirements.
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: add `the` after `read`.

File: src/xdocs/contributing.xml [hunk 8]
```diff
@@ -127,26 +127,27 @@
...
+         Small changes of configuration files, documentation fixes etc. can be contributed by starting
+         the commit message with one of "minor:", "config:", "infra:", "doc:" or "spelling:",
+         followed by a blank and a brief single-line message.
       </p>
 
       <p>
-        After you do Pull Request please recheck that where are no problems from our build systems:
-        <a href="https://travis-ci.org/checkstyle/checkstyle">Travis</a> and others ... .
-        You will see that your Pull Request on Github is marked by that jobs automatically in few minutes.
+        After submitting a pull request, it will be automatically checked by
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: `pull request` => `Pull Request`

File: src/xdocs/contributing.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 3 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 5 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 6 above)
Comment (romani) [reply src/xdocs/contributing.xml]: a bit extended.

File: src/xdocs/contributing.xml (hunk 7 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done

File: src/xdocs/contributing.xml (hunk 8 above)
Comment (romani) [reply src/xdocs/contributing.xml]: done, all cases changed

**Comment by romani** [comment]
@rnveach , please review,  I can not squash commits, history changes is forbidden for all branches in main repo, but I will do this on merge manually.

**Comment by rnveach** [comment]
> history changes is forbidden for all branches in main repo

Maybe you should block new branches in main repo :)

File: src/xdocs/contributing.xml [hunk 9]
```diff
@@ -127,26 +127,28 @@
...
 
       <p>
-        Please use <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
-        feature of <a href="https://github.com/">Github</a>.<br/>
-        Please provide wide description of update with detailed explanation of problem
-        and how you propose to resolve it.
-        <b>ATTENTION:</b> Please recheck that in your Pull Request there is ONLY your changes and
-        all of them are rebased from most recent HEAD of our master branch.
+        Please create a <a href="https://help.github.com/articles/using-pull-requests/">Pull Request</a>
+        for your contribution. In the Pull Request description, add any explanation of the implementation
```
Comment (rnveach) [inline src/xdocs/contributing.xml]: https://github.com/checkstyle/checkstyle/pull/5602#discussion_r174334339
`explanation` => `explanations`

Comment (romani) [reply src/xdocs/contributing.xml]: done.

**Comment by romani** [comment]
merged manually with squash

**Pull request closed without merging**